### DISCUSSION TITLE
Premières analyses des DUMPS - addressCountry & addressRegion

### DISCUSSION THREAD

**Discussion by GenFab28** [culturecreates/artsdata-planet-lavitrine#27] [Modelling discussions]
En important les DUMPS dans le système de La Vitrine, voici les observations sur certaines inconsistances / à discuter voir si on peut améliorer les importations

1. Certain addressCountry et addressRegion n’utilisent pas un code ISO mais un format texte  exemple:
"addressRegion": {
            "fr": "Québec"
          },
          
          **comme tous ces cas ne concerne que Québec et Canada nous avons effectué une correspondance mais cela pourrait être modifié dans les datas de base.**
          
2. Certaines locations ont deux streetAddress exemple : 

 "address": {
          "@id": "http://kg.artsdata.ca/resource/K11-22#PostalAddress",
          "type": "PostalAddress",
          "addressCountry": {
            "@none": "CA"
          },
          "addressLocality": {
            "@none": [
              "Montréal",
              "Montreal"
            ]
          },
          "addressRegion": {
            "@none": "QC"
          },
          "postalCode": "H2X 1Y9",
          "streetAddress": {
            "@none": [
              "260, boul. de Maisonneuve Ouest",
              "175 rue Saint-Catherine Ouest"
            ]
          }
        },

3. Certaine Place n’ont pas de Région :
"location": {
        "@id": "http://api.footlight.io/places/65dc68abbec01b00642c63ca",
        "type": "Place",
        "address": {
          "@id": "http://api.footlight.io/postalAddress/65dc68aabec01b00642c63be",
          "type": "PostalAddress",
          "addressCountry": {
            "fr": "Canada"
          },
          "addressLocality": {
            "fr": "Trois-Rivières"
          },
          "postalCode": "G9A 2G9",
          "streetAddress": {
            "fr": "375, rue des Forges"
          },
          "http://schema.org/addressRegion": null
        },
        "name": {
          "fr": "Le 507"
        },
        "sameAs": [],
        "latitude": null,
        "longitude": null,
        "containedInPlace": null
      },

@saumier   @christianroy

**Comment by saumier** [comment]
I have fixed some of these.

The second one with "Cinquième Salle" is very interesting.  ScenePro has the artist entrance 260, boul. de Maisonneuve Ouest, while other sources have the main public entrance at 175 rue Saint-Catherine Ouest. Artsdata keeps both and usually picks one, but this one was hard to pick because they are both correct but for different reasons. 

There are certainly more issues we will need to resolve. Until there are no validation errors.  We should discuss in our meeting tomorrow about ways to improve using the validation reports.

Good news is that these errors are **already being reported** in the validation report with each dump.

For example, if you take the link to download a dump like culture-mauricie from yesterday:
https://huginn-data.s3.ca-central-1.amazonaws.com/lavitrine/2024-05-29T02-22-06/culture-mauricie-cms-events.json

and you replace the ending with .yml you will get the dump validation report.
https://huginn-data.s3.ca-central-1.amazonaws.com/lavitrine/2024-05-29T02-22-06/culture-mauricie-cms-events.yml

The specific error in "Le 507" place missing schema:addressRegion that you mention is reported like this:
```
Result for: focus: <http://api.footlight.io/postalAddress/65dc68aabec01b00642c63be>
  path: schema:addressRegion
  shape: <http://kg.artsdata.ca/shacl/addressRegionProp>
  resultSeverity: shacl:Violation
  component: shacl:MinCountConstraintComponent
  message: "0 >= minCount 1"
```

It is also possible to view errors in a more human friends way using Nebula like the missing region for "Le 507" address:
http://kg.artsdata.ca/entity?uri=http%3A%2F%2Fapi.footlight.io%2FpostalAddress%2F65dc68aabec01b00642c63be
